api/components_users.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from chromestatus_openapi.models import (
+  ComponentsUsers,
+  ComponentsUsersComponentsInner,
```

Comment:
It's not your fault, but I don't like these generated model object names.  I just think that they will be hard to work with.  Is there an alternative style of using these where we create the overall response object early and then call type-safe methods to mutate it to build up the final response without all the mid-level objects being explicit in our code?  (That's one way to use protobufs.)

Reply:
I 100% agree that the current names are bad. But they can be changed. (Naming is always hard haha). Do you have any recommendations?

I thought about changing `ComponentsUsers` -> `ComponentsUsersResponse`, `ComponentsUsersComponentsInner` -> `OwnersAndSubscribersOfComponent` and `ComponentsUsersUsersInner` to `ComponentsUser`.


Could you give an example of what you mean for the style.

This is something that is currently possible but not sure if it meets what you want.

```py
resp = ComponentsUsers()

# build up users
users = []
# do stuff
resp.users(users)


# build up the components
# do stuff
resp.components(components[1:])


return resp.to_dict()
```

Reply:
Yeah, I think that might be a little more clear.  Also, I always prefer to use the assignment operator syntax if possible rather than the setter function syntax.

In your `# do stuff` section, you would still need to construct the inner objects and reference their classes by name.  Those inner class names are kind of long and we need all those imports.  

Protocol buffers can avoid that a bit having repeated fields offer an `.add()` method.    Here's an example from https://protobuf.dev/getting-started/pythontutorial/ that does not need to reference the PhoneNumber class by name.
```
import addressbook_pb2
person = addressbook_pb2.Person()
person.id = 1234
person.name = "John Doe"
person.email = "jdoe@example.com"
phone = person.phones.add()
phone.number = "555-4321"
phone.type = addressbook_pb2.Person.HOME
```

It can even forward the add() keyword parameters to the child object constructor, like:
`person.phones.add(number="555-4321", type=addresbook_pb2.Person.HOME)`
See https://protobuf.dev/reference/python/python-generated/

This allows for child objects to be added without mentioning their class names, which makes the application code more succinct.


In terms of naming the model classes, I like the idea of using the word Response and your other suggested changes.

Reply:
I have changed the class names in https://github.com/GoogleChrome/chromium-dashboard/pull/2742 and rebased this PR with those changes.

The `add` method is a nice feature. Thanks for the example and explanation! I can see the benefit of the one line `.add(all the kwargs)`.